lib/client/rpc/proposals.ts

```diff
@@ -242,7 +251,184 @@ export class FutarchyRPCProposalsClient implements FutarchyProposalsClient {
+  }
+
+  async getUserTokenBalance(tokenMintAddress: PublicKey) {
+    // Get the associated token account for the user's wallet and the token mint
+    const userTokenAccountAddress = await getAssociatedTokenAddress(
+      new PublicKey(tokenMintAddress),
+      this.rpcProvider.publicKey
+    );
+
+    // Fetch the account info
+    const accountInfo = await getAccount(this.rpcProvider.connection, userTokenAccountAddress);
+
+    // The amount is a BigInt, convert it to a string if necessary
+    return accountInfo.amount.toString();
+  }
+
+  async mintIx(amount: number,
+    vaultAccountAddress: PublicKey,
+    vaultAccount: VaultAccountWithProtocol) {
+    const ixs = [
+      createAssociatedTokenAccountIdempotentInstruction(
+        this.rpcProvider.publicKey,
+        getAssociatedTokenAddressSync(
+          vaultAccount.conditionalOnFinalizeTokenMint,
+          this.rpcProvider.publicKey,
+          true
+        ),
+        this.rpcProvider.publicKey,
+        vaultAccount.conditionalOnFinalizeTokenMint
+      ),
+      createAssociatedTokenAccountIdempotentInstruction(
+        this.rpcProvider.publicKey,
+        getAssociatedTokenAddressSync(
+          vaultAccount.conditionalOnRevertTokenMint,
+          this.rpcProvider.publicKey,
+          true
+        ),
+        this.rpcProvider.publicKey,
+        vaultAccount.conditionalOnRevertTokenMint
+      ),
+      await vaultAccount.protocol.vault.methods
+        .mintConditionalTokens(amount)
+        .accounts({
+          vault: vaultAccountAddress,
+          userConditionalOnFinalizeTokenAccount: getAssociatedTokenAddressSync(
+            vaultAccount.conditionalOnFinalizeTokenMint,
+            this.rpcProvider.publicKey,
+            true
+          ),
+          userConditionalOnRevertTokenAccount: getAssociatedTokenAddressSync(
+            vaultAccount.conditionalOnRevertTokenMint,
+            this.rpcProvider.publicKey,
+            true
+          ),
+          userUnderlyingTokenAccount: getAssociatedTokenAddressSync(
+            vaultAccount.underlyingTokenMint,
+            this.rpcProvider.publicKey,
+            true
+          ),
+          vaultUnderlyingTokenAccount: vaultAccount.underlyingTokenAccount,
+          conditionalOnFinalizeTokenMint:
+            vaultAccount.conditionalOnFinalizeTokenMint,
+          conditionalOnRevertTokenMint:
+            vaultAccount.conditionalOnRevertTokenMint,
+        })
+        .instruction(),
+    ];
+    return ixs
+  }
+
+  async createProposal(daoAggregate: DaoAggregate, user: PublicKey) {
+    const proposalKP = Keypair.generate();
+    const proposal = proposalKP.publicKey;
+
+    console.log(proposal)
+    const currentDao = daoAggregate.daos.filter((dao) => dao.protocol.autocrat.programId.toString() == this.autocratClient.autocrat.programId.toString())[1]
+    if (!currentDao) return
+
+    console.log(currentDao.publicKey.toString())
+    const autocrat = this.autocratClient.autocrat
```

Comment:
Since these are initialized already, wouldn't it be better to surface them at a higher order so you're not establishing the consts in each function? I agree that breaking it out makes sense. EXCEPT in the case where you want to build each program accordingly (as I'm not sure the CURRENT SDK is backwards compatible (eg you'll want to create conditional vault from anchor program with a different IDL).

Reply:
This goes for vaultClient, aamClient too

Reply:
Yes for now most of it should be from anchor as futarchy-ts only supports v1, should've cleaned it now but we look back at it

I'll clean it and take it away from clients so it can be accessed everywhere